### ISSUE TITLE
Accessibility issues with page navigation on News page

### ISSUE THREAD

**Issue by erosselli** [django/djangoproject.com#2507]
Hello! I noticed a couple of accessibility issues with the pagination navigation on the [News page](https://www.djangoproject.com/weblog/), shown on the screenshot below. 

<img width="335" height="126" alt="Pagination navigation on the News page, showing text saying 'Page 1 of 84' and a button with an arrow to go to the next pag" src="https://github.com/user-attachments/assets/b9763319-d81b-4517-be37-c8f159fe564c" />

I think this would be a good first issue for one of the Djanognauts in the Djangonaut Space Session 6 Team Mars, which will be focused on accessibility improvements both for Django core and for djangoproject.com (I will be the Navigator for this team). 

#### Semantic HTML 

The markup for this section does not use semantic HTML, since it's using a `<li>` element within a parent that does not have a list role (since the `<ul>` element has `role="navigation"`). I think we should probably replace the `<ul>` and `<li>` elements with semantically appropriate ones. 

Current HTML: 
```
<ul class="nav-pagination" role="navigation">
      <span class="page-current">
        Page 1 of 84
      </span>
        <li>
               <a rel="next" class="next" href="?page=2"> ... </a>
         </li>
    </ul>
```

#### Screen reader support 
When using a screen reader to navigate through the links,  the "Page 1 out of XX" text isn't read as part of the link   context. I tested on VoiceOver and just got "Link, next, navigation". I think making it so that the page number and total also get read by VoiceOver would probably be a better UX.

**Comment by mahicodes28** [comment]
heyyyy @erosselli , I’ve opened a PR addressing this issue: #2516

**Comment by MANAS225** [comment]
Hi @erosselli 

I’d like to work on this accessibility issue. I’ll start by reviewing the current pagination markup and improving the semantic HTML structure (e.g., replacing improper <ul> usage with appropriate <nav> and list elements), and then address the screen reader support using proper ARIA attributes.

Could you please assign this issue to me? I’ll share the proposed changes in a PR for feedback.

Thank you!

**Comment by smallsaucepan** [comment]
Also worth noting pagination accessibility is an issue on [search result](https://docs.djangoproject.com/en/6.1/search/?q=django&category=) and [community feed](https://www.djangoproject.com/community/blogs/) pages.

Is still being held aside for a Djangonaut project?

**Comment by erosselli** [comment]
> Also worth noting pagination accessibility is an issue on [search result](https://docs.djangoproject.com/en/6.1/search/?q=django&category=) and [community feed](https://www.djangoproject.com/community/blogs/) pages.
> 
> Is still being held aside for a Djangonaut project?


Hello @smallsaucepan , thanks for pointing out that the issue happens in other pages too! This issue is no longer behind held aside for Djangonaut Space, though there is an open PR for it already.

**Comment by smallsaucepan** [comment]
Thanks for clarifying. Will add a few suggestions to the existing PR.
